### PR TITLE
Add KernelInfo tests

### PR DESCRIPTION
**Pull request by Watson1978**
This patch will improve coverage of rmkinfo.c

* Before
  - Line coverage : 24.5 %
  - Functions : 30.0 %

* After
  - Line coverage : 97.4 %
  - Functions : 100.0 %

Related to https://github.com/rmagick/rmagick/issues/542

### CODE REVIEW

File: test/KernelInfo.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        assert_raise(RuntimeError) { Magick::KernelInfo.new(k) }
+      else
+        assert_instance_of(Magick::KernelInfo, Magick::KernelInfo.new(k))
+      end
+    end
+    assert_raise(RuntimeError) { Magick::KernelInfo.new('') }
+    assert_raise(TypeError) { Magick::KernelInfo.new(42) }
+  end
+
+  def test_zero_nans
```
Comment (dlemstra) [inline test/KernelInfo.rb]: Maybe remove this tests? It fails with ruby 2.3 on Windows and this method is deprecated and will be removed soon.

File: test/KernelInfo.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    assert_nil(@kernel.zero_nans)
+  end
+
+  def test_unity_add
+    assert_nil(@kernel.unity_add(1.0))
+    assert_nil(@kernel.unity_add(12))
+    assert_raise(TypeError) { @kernel.unity_add('x') }
+  end
+
+  def test_show
```
Comment (dlemstra) [inline test/KernelInfo.rb]: This method is deprecated and will be removed soon. Think we can remove it.

Comment (mockdeep) [reply test/KernelInfo.rb]: I think ideally while we have code we're supporting, we should have it under test, even if we intend to remove it in the near future.

File: test/KernelInfo.rb (hunk 1 above)
Comment (mockdeep) [reply test/KernelInfo.rb]: Looks like it's causing a seg fault.

Comment (Watson1978) [reply test/KernelInfo.rb]: @dlemstra NOT agreed.
When we will totally stop to support them, then we will just remove unnecessary tests.

File: test/KernelInfo.rb (hunk 2 above)
Comment (Watson1978) [reply test/KernelInfo.rb]: @dlemstra NOT agreed.

File: test/KernelInfo.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    assert_nil(@kernel.show)
+  end
+
+  def test_scale
```
Comment (Watson1978) [inline test/KernelInfo.rb]: Seems this test causes SEGV with Ruby 2.3 on Windows

File: test/KernelInfo.rb (hunk 1 above)
Comment (dlemstra) [reply test/KernelInfo.rb]: I just think that you should not spend to much of your time on this. It is very unlikely that someone is using this method.

File: test/KernelInfo.rb (hunk 3 above)
Comment (dlemstra) [reply test/KernelInfo.rb]: I wonder if this is actually a bug in ruby 2.3 on Windows that this code is hitting. Could you get a backtrace with a debugger so we know where the segfault is raised?

File: test/KernelInfo.rb (hunk 1 above)
Comment (Watson1978) [reply test/KernelInfo.rb]: It's not a job, so I think we do whatever you like.  Now I want to improve test for the future ;)

File: test/KernelInfo.rb (hunk 3 above)
Comment (Watson1978) [reply test/KernelInfo.rb]: I want to know how to debug on Windows...

**Comment by Watson1978** [comment]
Ruby 2.3 environment in AppVeyor, it uses gcc 4.7.2.
スクリーンショット 2019-05-01 4 10 13

Seems the gcc generate wrong binary for non-declaration function...
When call ImageMagick private API, wrong value will be passed into that.

So, It may crash if user use the old gcc. it might be regardless of platform.

**Pull request merged**